contracts/Core/JobManager.sol

```diff
@@ -37,79 +38,209 @@ contract JobManager is ACL, JobStorage {
-            epoch,
-            url,
-            selector,
-            name,
-            repeat,
-            msg.sender,
-            msg.value,
-            block.timestamp
+            numAssets, 
+            epoch, 
+            url, 
+            selector, 
+            name, 
+            repeat, 
+            msg.sender, 
+            msg.value, 
+            block.timestamp, 
+            uint256(assetTypes.Job)
         );
+        // jobs.push(job);
     }
 
-    function fulfillJob(
-        uint256 jobId,
-        uint256 value
-    )
+    function fulfillAsset(
+        uint256 id, 
+        uint256 value   
+    ) 
         external 
         onlyRole(Constants.getJobConfirmerHash())
     {
-        Structs.Job storage job = jobs[jobId];
         uint256 epoch = stateManager.getEpoch();
+        if(jobs[id].assetType==uint256(assetTypes.Job)){
+
+            Structs.Job storage job = jobs[id];
+
+            if (!job.repeat) {
+                job.fulfilled = true;
+            }
 
-        if (!job.repeat) {
-            job.fulfilled = true;
+            job.result = value;
+            emit JobReported(
+                job.id,
+                value,
+                epoch,
+                job.url,
+                job.selector,
+                job.name,
+                job.repeat,
+                job.creator,
+                job.credit,
+                job.fulfilled,
+                block.timestamp
+            );
         }
+        else if(collections[id].assetType==uint256(assetTypes.Collection)){
 
-        job.result = value;
-        emit JobReported(
-            job.id,
-            value,
+            Structs.Collection storage collection = collections[id];
+
+            collection.result = value;
+
+            emit CollectionReported(
+                collection.id,
+                value,
+                epoch,
+                collection.name,
+                collection.aggregationMethod,
+                collection.jobIDs,
+                collection.creator,
+                collection.credit,
+                block.timestamp
+            );
+        }
+    }
+
+    function createCollection(
+        string calldata name, 
+        uint256[] memory jobIDs,
+        uint32 aggregationMethod
+    ) external payable 
+    {
+        require(aggregationMethod < 3,"Aggregation range out of bounds");
+
+        numAssets++;
+        uint256 epoch = stateManager.getEpoch();
+        collections[numAssets].id = numAssets; 
+        collections[numAssets].name = name; 
+        collections[numAssets].aggregationMethod = aggregationMethod; 
+        collections[numAssets].epoch = epoch;
+        collections[numAssets].creator = msg.sender;
+        collections[numAssets].credit = msg.value;
+        for(uint256 i = 0; i < jobIDs.length; i++){
+            require(jobs[jobIDs[i]].assetType==uint256(assetTypes.Job),"Job ID not present");
+            if(collections[numAssets].jobIDExist[jobIDs[i]]){
```

Comment:
Hey, What this line intends to do ?

Reply:
Cant we handle this using require statement only

Reply:
This condition is there to handle any duplicates inside the jobIDs array. Yes, require can be used here but i thought this should also work

Reply:
Wouldn't require statement revert the whole tx if the condition is false? I was under the impression that `if` is used here because the loop should `continue` & skip the rest of the computation if Job exists in collection. @SamAg19